packages/puppeteer-core/src/api/ElementHandle.ts

```diff
@@ -861,6 +873,35 @@ export abstract class ElementHandle<
     options?: Readonly<KeyPressOptions>
   ): Promise<void>;
 
+  async #clickableBox(): Promise<BoundingBox | null> {
+    const adoptedThis = await this.frame.isolatedRealm().adoptHandle(this);
+    const box = await adoptedThis.evaluate(element => {
+      if (!(element instanceof Element)) {
+        return null;
+      }
+      return element.getClientRects()[0]?.toJSON() as DOMRect;
```

Comment:
also we checked that rect is in the viewport. Do we not need that?

Reply:
FWIU, the check is only kind of useful in the case where:

 1. the element is not in an iframe,
 2. overflow is hidden so scrolling to the element is not possible
 
I suspect that (2) is not very common.

Reply:
What happens if we click on a portion of the element that is outside of the viewport? Imagine we have a long paragraph of text in one element that starts before the viewport and ends after. What happens if we click on the rect for the first line that is outside of the viewport?

Reply:
Done.